File: Jump-Game/Jump-Game.py

class Solution:
    # @param A, a list of integers
    # @return a boolean
    def canJump(self, A):
        ans=0
        for i in range(len(A)):
            if ans<i:
                return False
            ans=max(ans,i+A[i])
            if ans>=len(A)-1:
                return True
        return ans>=len(A)-1
 

File: Jump-Game/Jump-Game.cpp

class Solution {
public:
    bool canJump(int A[], int n) {
        int ans=0;
        for(int i=0;i<n;i++)
        {
            if(i>ans)
            {
                return false;
            }
            ans=max(ans,i+A[i]);
            if(ans>=n-1)
            {
                return true;
            }
        }
        return ans>=n-1;
    }
};


 